inc/components/class-menu.php

```diff
@@ -49,8 +61,55 @@ public function parse_wp_menu_by_location( string $menu_location ) {
 	public function parse_wp_menu( string $menu_slug ) {
 		$menu_items = wp_get_nav_menu_items( $menu_slug );
 
+		$this->children = $this->build_menu( $menu_items );
+
 		return $this;
 	}
+
+	/**
+	 * Recursive function to build a complete menu with children menu items.
+	 *
+	 * @param  array   $menu_items Menu items.
+	 * @param  integer $parent_id  Parent menu ID.
+	 * @return array All menu items.
+	 */
+	public function build_menu( $menu_items, $parent_id = 0 ) {
+
+		// Start building menu.
+		$menu = [];
+
+		// Loop through all menu items.
+		foreach ( (array) $menu_items as $key => $menu_item ) {
+
+			// Current menu's id.
+			$menu_item_id = $menu_item->ID;
+
+			// Current menu's parent id.
+			$menu_item_parent_id = absint( $menu_item->menu_item_parent );
+
+			// Is the current menu item a child of the parent item.
+			if ( $menu_item_parent_id === $parent_id ) {
+
+				// Get parsed menu item.
+				$clean_menu_item = menu_item()->parse_menu_post( $menu_item );
```

Comment:
if it's not rendered moot by my previous comment, writing functions with very generic names like `menu_item()` (even though it's namespaced) makes for confusing code. I had to dig around to see where that was defined, assuming it was a WP core function. Same comment applies to `menu()`

Reply:
It's a helper function for getting an instance of the menu_item component. I'm not sure what else would make sense here. I could shift to a pattern more like `( new \WP_Irving\Component\Menu_Item() )->function_call()` and there's nothing here that precludes that from being used, but I didn't see this being a blocker, so we moved forward with it. Happy to refactor moving forward if there's a strong argument to be made about